src/blog/_posts/2018-05-09-opera-53.md

```diff
@@ -0,0 +1,234 @@
+
+The new rendering context allows for more efficient rendering of an image to a
+canvas. Instead of creating an `<img>` element and rendering it to the canvas,
+which causes multiple copies of the image to be stored in memory, an
+`ImageBitmap` object can now be displayed in the canvas directly, effectively by
+transferring ownership of the pixel data to the canvas element.
+
+    const image = await createImageBitmap(imageBlob);
+    const canvas = document.createElement('canvas');
+    const context = canvas.getContext('bitmaprenderer');
+    context.transferFromImageBitmap(image);
+
+Since the source image can be fetched and decoded asynchronously, this can be
+used in e.g. a WebGL game to load textures on the fly, without jank.
+
+
+## CSS Typed Object Model
+
+The [CSS Typed Object Model](https://drafts.css-houdini.org/css-typed-om/)
+exposes the values of CSS properties to scripts as typed JavaScript objects,
+instead of as CSS formatted strings. This makes it more straight-forward to
+write code that reads and manipulates CSS properties, and the code will
+typically execute faster.
+
+Opera 53 implements the object model for a [subset of CSS properties](https://chromium.googlesource.com/chromium/src/+/master/third_party/blink/renderer/core/css/cssom/README.md).
+
+
+## Asynchronous clipboard API
+
+A new asynchronous clipboard API has been added that is simpler to use than the
+old `document.execCommand("copy")` API, and that integrates with the
+[Permissions API](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API).
+
+With this new API, copying some text to the clipboard is as simple as
+
+    try {
+        await navigator.clipboard.writeText("some text");
+    } catch {
+        console.error("Could not copy text to the clipboard!");
+    }
+
+and reading text from the clipboard is equally easy:
+
+    const text = await navigator.clipboard.readText();
+    console.log("Clipboard text:", text);
+
+For more information about the new API and how to use it, see [Unblocking
+Clipboard Access](https://developers.google.com/web/updates/2018/03/clipboardapi).
+
+
+## AudioWorklet
+
+The [`AudioWorklet`](https://developers.google.com/web/updates/2017/12/audio-worklet)
+API provides a synchronous JavaScript execution context that allows developers
+to programmatically control audio output with lower latency and higher stability
+than the legacy `ScriptProcessorNode` API, which will be deprecated eventually.
+
+Demos and other `AudioWorklet` resources can be found at [Google Chrome
+Labs](https://googlechromelabs.github.io/web-audio-samples/audio-worklet/).
+
+
+## Other web platform features in this release
+
+### Blink > Animation
+
+* To maintain compatibility with other implementations, the `add` and
+  `accumulate` keywords will not throw errors. They will soon be supported by
+  Opera.
+
+### Blink > CSS
+
+* The expressions `calc()`, `min()` and `max()` are now supported in media
```

Comment:
AFAIK, only calc() is supported (max() and min() haven't been implemented at all), so this could probably be rephrased as "calc() expressions are now supported..." or so.

Reply:
Yes, `min()`/`max()` appear to be unsupported. Removed them. But `calc()` works in O53 and not in O52.